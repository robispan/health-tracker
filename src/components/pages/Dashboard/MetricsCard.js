import React from 'react';

import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';

const MetricsCard = ({ ratio, label }) => {
  return (
    <div className='metrics-card'>
      <h3 className='metrics-card-title center'>{label}</h3>
      {ratio ? (
        <CircularProgressbar
          value={ratio}
          text={`${ratio}%`}
          background
          backgroundPadding={6}
          styles={buildStyles({
            backgroundColor: '#00b9bb',
            textColor: '#fff',
            pathColor: '#fff',
            trailColor: 'transparent'
          })}
        />
      ) : (
        <div className='center'>
          Connect trackers on homepage to see your progression!
        </div>
      )}
    </div>
  );
};

export default MetricsCard;
