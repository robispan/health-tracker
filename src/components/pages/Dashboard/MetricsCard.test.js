import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import MetricsCard from './MetricsCard';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(<MetricsCard ratio={70} label='Heart rate' />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders without ratio prop', () => {
  const { asFragment } = render(<MetricsCard label='Heart rate' />);
  expect(asFragment()).toMatchSnapshot();
});

it('inserts label to title', () => {
  const { getByText } = render(<MetricsCard label='Heart rate' />);
  expect(getByText('Heart rate'));
});
