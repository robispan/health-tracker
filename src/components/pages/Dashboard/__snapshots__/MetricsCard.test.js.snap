// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="metrics-card"
  >
    <h3
      class="metrics-card-title center"
    >
      Heart rate
    </h3>
    <svg
      class="CircularProgressbar "
      data-test-id="CircularProgressbar"
      viewBox="0 0 100 100"
    >
      <circle
        class="CircularProgressbar-background"
        cx="50"
        cy="50"
        r="50"
        style="fill: #00b9bb;"
      />
      <path
        class="CircularProgressbar-trail"
        d="
      M 50,50
      m 0,-40
      a 40,40 0 1 1 0,80
      a 40,40 0 1 1 0,-80
    "
        fill-opacity="0"
        stroke-width="8"
        style="stroke: transparent; stroke-dasharray: 251.32741228718345px 251.32741228718345px; stroke-dashoffset: 0px;"
      />
      <path
        class="CircularProgressbar-path"
        d="
      M 50,50
      m 0,-40
      a 40,40 0 1 1 0,80
      a 40,40 0 1 1 0,-80
    "
        fill-opacity="0"
        stroke-width="8"
        style="stroke: #fff; stroke-dasharray: 251.32741228718345px 251.32741228718345px; stroke-dashoffset: 75.39822368615505px;"
      />
      <text
        class="CircularProgressbar-text"
        style="fill: #fff;"
        x="50"
        y="50"
      >
        70%
      </text>
    </svg>
  </div>
</DocumentFragment>
`;

exports[`renders without ratio prop 1`] = `
<DocumentFragment>
  <div
    class="metrics-card"
  >
    <h3
      class="metrics-card-title center"
    >
      Heart rate
    </h3>
    <div
      class="center"
    >
      Connect trackers on homepage to see your progression!
    </div>
  </div>
</DocumentFragment>
`;
